LaunchDarkly/LaunchDarkly/ServiceObjects/ClientServiceFactory.swift

```diff
@@ -121,8 +125,12 @@ final class ClientServiceFactory: ClientServiceCreating {
         DiagnosticCache(sdkKey: sdkKey)
     }
 
-    func makeDiagnosticReporter(service: DarklyServiceProvider, environmentReporter: EnvironmentReporting) -> DiagnosticReporting {
-        DiagnosticReporter(service: service, environmentReporting: environmentReporter)
+    func makeDiagnosticReporter(config: LDConfig, service: DarklyServiceProvider, environmentReporter: EnvironmentReporting) -> DiagnosticReporting {
+        if config.sendEvents && !config.diagnosticOptOut {
```

Comment:
The suspense between the two comments. I read the first one and I was like, "yes, but that indirectness scares me".

Reply:
😆 yes, I hate that there are even two paths like this. But there are protocols that are depending on others in a cycle and so really teasing that out as much as I would prefer is probably more effort than it is worth. Especially since we're pushing for a larger re-work.